test/python/circuit/test_compose.py

```diff
@@ -592,6 +593,64 @@ def test_wrapping_unitary_circuit(self):
             qc = qc_init.compose(qc_nonunitary, wrap=True)
             self.assertIsInstance(qc.data[1][0], Instruction)
 
+    def test_single_bit_condition(self):
+        """Test that compose can correctly handle circuits that contain conditions on single
+        bits.  This is a regression test of the bug that broke qiskit-experiments in gh-7653."""
+        base = QuantumCircuit(1, 1)
+        base.x(0).c_if(0, True)
+        test = QuantumCircuit(1, 1).compose(base)
+        self.assertIsNot(base.clbits[0], test.clbits[0])
+        self.assertEqual(base, test)
+        self.assertIs(test.data[0][0].condition[0], test.clbits[0])
+
+    def test_condition_mapping_ifelseop(self):
+        """Test that the condition in an `IfElseOp` is correctly mapped to a new set of bits and
+        registers."""
+        base_loose = Clbit()
+        base_creg = ClassicalRegister(2)
+        base_qreg = QuantumRegister(1)
+        base = QuantumCircuit(base_qreg, [base_loose], base_creg)
+        with base.if_test((base_loose, True)):
+            base.x(0)
+        with base.if_test((base_creg, 3)):
+            base.x(0)
+
+        test_loose = Clbit()
+        test_creg = ClassicalRegister(2)
+        test_qreg = QuantumRegister(1)
+        test = QuantumCircuit(test_qreg, [test_loose], test_creg).compose(base)
+
+        bit_instruction, *_ = test.data[0]
+        reg_instruction, *_ = test.data[1]
+        self.assertIs(bit_instruction.condition[0], test_loose)
```

Comment:
This is a bit weird to me. I assume you're doing this to do exact instance matching so `test_loose is bit_instruction.condition`. This was just a bit surprising to me, because I it's very uncommon to do it this way.

Reply:
Yeah, it's specifically a test that a condition on a loose bit is correctly converted to a condition on a new loose bit as part of the `compose` operation.  The logical condition that `bit_instruction` refers to was created with `base_loose`, but since it went through the composition, it now needs to be `test_loose` because `base_loose` isn't in the new circuit.  There's a little bit of weirdness around equality tests with old-style and new-style bits, so I just wanted to be super explicit.